lib/practice_exercises.rb

```diff
@@ -1,13 +1,31 @@
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n)
+# Space Complexity: O(n)
 def is_palindrome(string)
-  raise NotImplementedError, "Not implemented yet"
+  alphanumeric_chars = string.gsub(/[^a-z^0-9]/i, '').downcase.split("")
+  i = 0
+  j = alphanumeric_chars.length - 1
+  while i < j
+    return false unless alphanumeric_chars[i] == alphanumeric_chars[j]
+    i += 1
+    j -= 1
+  end
+
+  return true
 end
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: 0(n)
+# Space Complexity: O(1)
 def longest_prefix(strings)
```

Comment:
👍  Technically since you're returning a new string (the prefix), worst case this method has O(n) space complexity where n is the size of the smallest string.